### PR TITLE
[DOCS] Edit JVM settings info. Closes #72259

### LINKED ISSUE #72259: [DOCS] Make JVM options docs more scannable
Hey team! Following up on a discussion that was had around the ambiguity of file extensions for jvm options files placed in jvm.options.d here: 
https://www.elastic.co/guide/en/elasticsearch/reference/current/jvm-options.html#jvm-options
https://www.elastic.co/guide/en/elasticsearch/reference/current/advanced-configuration.html#setting-jvm-heap-size

Here is what we say in the jvm.options doc above

> When installing from the tar or zip distributions, the root jvm.options configuration file is config/jvm.options and custom JVM options files can be added to config/jvm.options.d/.

This makes no mention that any file places in that directory (eg `foo`) would need to be named `foo.options` in order to be picked up and used by Elasticsearch. A quick example to show that the files in the jvm.options.d directory should explicitly have the `.options` extension would help a great deal.

Thank you!

### PR DESCRIPTION
**Pull request by debadair**
Called out the naming requirements for the JVM options files in the section about setting the heap size, edited the JVM options topic, and consolidated the info onto a single page. Also moved the "Important" settings topic into the TOC position where the JVM options topic used to be.

### CODE REVIEW

File: docs/reference/setup/jvm-options.asciidoc [hunk 1]
```diff
@@ -1,90 +1,83 @@
...
+`ES_JAVA_OPTS` in the <<sysconfig,system configuration file>>.
 
-Additionally, some other Java programs support the `JAVA_OPTS` environment
-variable. This is *not* a mechanism built into the JVM but instead a convention
-in the ecosystem. However, we do not support this environment variable, instead
-supporting setting JVM options via the `jvm.options` file or the environment
-variable `ES_JAVA_OPTS` as above.
+NOTE: {es} ignores the `JAVA_TOOL_OPTIONS` and `JAVA_OPTS` environment variables.
+You must configure custom JVM options files or the `ES_JAVA_OPTS`
+environment variable to override the default JVM options.
```
Comment (jrodewig) [inline docs/reference/setup/jvm-options.asciidoc:83]: I don't think the last sentence is needed. We state this earlier.
```suggestion
NOTE: {es} ignores the `JAVA_TOOL_OPTIONS` and `JAVA_OPTS` environment variables.
```

File: docs/reference/setup/jvm-options.asciidoc [hunk 2]
```diff
@@ -1,90 +1,83 @@
...
 
-An alternative mechanism for setting Java Virtual Machine options is via the
-`ES_JAVA_OPTS` environment variable. For instance:
+[[jvm-options-env]]
+===== Using environment variables to set JVM options
+
+In production, you should use JVM options files if you need to override the
+default settings. However, you can also set the JVM options through the
+ `ES_JAVA_OPTS` environment variable, which can be useful for development and
+ testing. For example:
```
Comment (jrodewig) [inline docs/reference/setup/jvm-options.asciidoc:70]: I'd make these sentences mirror each other a bit more.
```suggestion
In production, you should use JVM options files if to override default JVM
settings. In testing and development environments, you can also set the JVM
options through the `ES_JAVA_OPTS` environment variable.
```

File: docs/reference/setup/jvm-options.asciidoc [hunk 3]
```diff
@@ -1,90 +1,83 @@
...
--------------------------------------
+Where you put the JVM options files depends on the type of installation:
+
+* tar.gz or .zip: Add custom JVM options files to `config/jvm.options.d/`.
+* Debian or RPM: Add custom JVM options files to `/etc/elasticsearch/jvm.options.d/`.
+* Docker: Bind mount custom JVM options files into
+`/usr/share/elasticsearch/config/jvm.options.d/`.
+
+NOTE: Always add custom JVM options files instead of modifying the root
+`jvm.options` file.
```
Comment (jrodewig) [inline docs/reference/setup/jvm-options.asciidoc:19]: We can be more direct here.
```suggestion
NOTE: Do not modify the root `jvm.options` file. Use JVM options files instead.
```

**Review by jrodewig — APPROVED** [review]
LGTM. I left a few suggestions, but most of those are non-blocking nits. Feel free to ignore if wanted. Thanks for taking care of this @debadair.

File: docs/reference/setup/advanced-configuration.asciidoc [hunk 4]
```diff
@@ -1,55 +1,63 @@
...
 ----
 
-* Set `Xmx` and `Xms` to no more than the threshold for zero-based
+* Set `Xms` and `Xmx` to no more than the threshold for zero-based
 compressed oops. The exact threshold varies but 26GB is safe on most
 systems and can be as large as 30GB on some systems. You can verify that
 you are under this threshold by starting {es} with the JVM options
-`-XX:+UnlockDiagnosticVMOptions -XX:+PrintCompressedOopsMode` and looking for
-a line like the following:
+`-XX:+UnlockDiagnosticVMOptions -XX:+PrintCompressedOopsMode` and checking
```
Comment (jrodewig) [inline docs/reference/setup/advanced-configuration.asciidoc:51]: I saw https://github.com/elastic/elasticsearch/issues/71644 and tested this locally. It looks like `-XX:+UnlockDiagnosticVMOptions -XX:+PrintCompressedOopsMode` is no longer supported.

We may want to update this section based on that ticket. We can also handle as a separate issue if wanted.

**Review by pugnascotia — APPROVED** [review]
Looks good from my POV. Nothing to add to @jrodewig's feedback.

File: docs/reference/setup/advanced-configuration.asciidoc (hunk 4 above)
Comment (debadair) [reply docs/reference/setup/advanced-configuration.asciidoc:51]: I'll handle as a follow up.

File: docs/reference/setup/jvm-options.asciidoc [hunk 5]
```diff
@@ -1,90 +1,83 @@
...
+
+[[jvm-options-syntax]]
+===== JVM options syntax
+
+A JVM options file contains a line-delimited list of JVM arguments.
+Arguments are preceded by a dash (`-`).
+To apply the setting to specific versions, prepend the version
+or a range of versions followed by a colon.
```
[the remark this answers is not in the document]
Comment (debadair) [reply docs/reference/setup/jvm-options.asciidoc:27]: I think it makes sense to describe the general pattern before the specific examples.

File: docs/reference/setup/jvm-options.asciidoc [hunk 6]
```diff
@@ -1,90 +1,83 @@
...
 
-* lines beginning with a `-` are treated as a JVM option that applies
-  independent of the version of the JVM
+* Apply setting to all versions. For example:
```
[the remark this answers is not in the document]
Comment (debadair) [reply docs/reference/setup/jvm-options.asciidoc:27]: I tweaked it slightly differently to reduce the repetition.

File: docs/reference/setup/jvm-options.asciidoc [hunk 7]
```diff
@@ -1,90 +1,83 @@
...
 [source,text]
 -------------------------------------
-8-9:-Xmx2g
+8-:-Xmx2g
 -------------------------------------
 
-* all other lines are rejected
+Blank lines are ignored. Lines beginning with `#` are treated as comments
+and ignored. Lines that aren't commented out and aren't recognized
+as JVM arguments are rejected.
```
[the remark this answers is not in the document]
Comment (debadair) [reply docs/reference/setup/jvm-options.asciidoc:60]: This was deliberate to put the focus on the contents of the config file. I added that unrecognized args result in an error/failure to start.

File: docs/reference/setup.asciidoc [hunk 8]
```diff
@@ -41,7 +41,9 @@ include::setup/install.asciidoc[]
 
 include::setup/configuration.asciidoc[]
 
-include::setup/jvm-options.asciidoc[]
+include::setup/important-settings.asciidoc[]
+
+include::setup/sysconfig.asciidoc[]
```
Comment (debadair) [inline docs/reference/setup.asciidoc:46]:
```suggestion
```

File: docs/reference/setup.asciidoc [hunk 9]
```diff
@@ -97,10 +99,6 @@ include::settings/notification-settings.asciidoc[]
 
 include::setup/advanced-configuration.asciidoc[]
```
Comment (debadair) [inline docs/reference/setup.asciidoc:101]:
```suggestion
include::setup/advanced-configuration.asciidoc[]
include::setup/sysconfig.asciidoc[]
```

**Pull request merged**